enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.v1.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  SdkIframeEmbedSettings,
+  SdkIframeEmbedTagMessage,
+  SdkIframeEmbedTagSettings,
+} from "./types/embed";
+
+const EMBEDDING_ROUTE = "embed/sdk/v1";
+
+type EmbedSettingKey = keyof SdkIframeEmbedSettings;
+
+const ALLOWED_EMBED_SETTING_KEYS = [
+  "apiKey",
+  "instanceUrl",
+  "dashboardId",
+  "questionId",
+  "template",
+  "theme",
+  "locale",
+] as const satisfies EmbedSettingKey[];
+
+type AllowedEmbedSettingKey = (typeof ALLOWED_EMBED_SETTING_KEYS)[number];
+
+class MetabaseEmbed {
+  static readonly VERSION = "1.0.0";
+
+  private _settings: SdkIframeEmbedTagSettings;
+  private _isEmbedReady: boolean = false;
+  private iframe: HTMLIFrameElement | null = null;
+
+  constructor(settings: SdkIframeEmbedTagSettings) {
+    if (!settings.apiKey || !settings.instanceUrl) {
+      raiseError("api key and instance url must be provided");
+    }
+
+    this._validateEmbedSettings(settings);
+    this._settings = settings;
+    this._settings._isLocalhost = this.getIsLocalhost();
+
+    this._handleMessage = this._handleMessage.bind(this);
```

Comment:
I'm 50/50 we don't need this. The reason being that 1. `_handleMessage` is defined using arrow function, so `this` is already bound. But I'm not sure if this will take effect immediately in the constructor since `_setup` also calls `_handleMessage`

Can you try removing this line to see if it's still working as expected?

Reply:
Fixed in https://github.com/metabase/metabase/pull/56915/commits/b3edbee70938f4c5feb6f44be7dc4754f1b21448, thanks! I initially wrote it as a regular function and I must've moved it to arrow function later without cleaning this up.